docs/source/notes/amp_examples.rst

```diff
@@ -0,0 +1,158 @@
+.. _amp-examples:
+
+Automatic Mixed Precision Examples
+==================================
+
+.. currentmodule:: torch.cuda.amp
+
+.. contents:: :local:
+
+.. _gradient-scaling-examples:
+
+Gradient Scaling
+^^^^^^^^^^^^^^^^
+
+Gradient scaling helps ensure convergence when training with mixed precision,
+as explained :ref:`here<gradient-scaling>`.
+
+The following code snippets show how :class:`torch.cuda.amp.AmpScaler` can be used to
+perform dynamic gradient scaling automatically.
+
+Working with a Single Optimizer
```

Comment:
Per above comments, need more context before this. Why are we talking about optimizers at all? What is the e2e perspective here?